PowerValidatedSolutions.psm1

```diff
@@ -51379,46 +51378,134 @@ Function New-AriaNetworksvCenterDataSource {
     #>
 
     Param (
-        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
-        [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$username,
-		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$password,
-		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
-		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
-		[Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$fqdn,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$username,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$password,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$nickname,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$collectorId,
+      [Parameter (Mandatory = $true)] [ValidateNotNullOrEmpty()] [String]$enabled
     )
 
-		if ( -not $PsBoundParameters.ContainsKey("fqdn") -or ( -not $PsBoundParameters.ContainsKey("username") -or ( -not $PsBoundParameters.ContainsKey("password") -or ( -not $PsBoundParameters.ContainsKey("nickname") -or ( -not $PsBoundParameters.ContainsKey("collectorId") -or ( -not $PsBoundParameters.ContainsKey("enabled"))))))){
-			Write-Error "Please Provide the necessary parameters and try again"
-		}
-
     Try {
         $uri = "https://$ariaNetworksAppliance/api/ni/data-sources/vcenters"
-        $body = '{
-		"fqdn": "'+ $fqdn +'",
-		"proxy_id": "'+ $collectorId +'",
-		"nickname": "'+ $nickname +'",
-		"enabled": '+ $enabled +',
-		"credentials": {
-			"username": "'+ $username +'",
-			"password": "'+ $password +'"
-		},
-		  "ipfix_request": {
-			"enable_all": true
-		  }
-		}'
-
-        if ($PSEdition -eq 'Core') {
-            $DataSourceResponse = Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body -SkipCertificateCheck # PS Core has -SkipCertificateCheck implemented, PowerShell 5.x does not
-		} else {
-				$DataSourceResponse = Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body
-				Write-Output "Successfully added $vcenter to VMware Aria Operations for Networks as a Data Source"
-		}
+        $body = @{
+            fqdn               = $fqdn
+            proxy_id           = $collectorId
+            nickname           = $nickname
+            enabled            = $enabled
+            credentials        = @{
+              username = $username
+              password = $password
+            }
+            ipfix_request      = @{
+              "enable_all"= $true
+            }
+          } | ConvertTo-Json
+        Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body -SkipCertificateCheck
   } Catch {
         Write-Error $_.Exception.Message
   }
 }
 Export-ModuleMember -Function New-AriaNetworksvCenterDataSource
```

Comment:
```suggestion
        $body = @{
            fqdn = $fqdn
            proxy_id = $collectorId
            nickname = $nickname
            enabled = $enabled
            credentials = @{
                username = $username
                password = $password
            }
            ipfix_request = @{
                "enable_all"= $true
            }
        } | ConvertTo-Json
        Invoke-RestMethod -Uri $uri -Method 'POST' -Headers $ariaNetworksHeader -Body $body -SkipCertificateCheck
    } Catch {
        Write-Error $_.Exception.Message
    }
}
```

Reply:
Resolved in https://github.com/vmware/power-validated-solutions-for-cloud-foundation/commit/954c74bc51b1fef11dc26b704cc60d52dd1bba80